composition-js/src/validate.ts

```diff
@@ -720,13 +730,29 @@ class ValidationTraversal {
       subgraphNameToGraphEnumValue,
     );
   }
+  
+  pushStack(state: ValidationState) {
+    this.totalValidationSubgraphPaths += state.subgraphPathInfos.length;
```

Comment:
I think that the number of `ValidationState` is correlated to`state.supergraphPath.nextEdges()`, than `subgraphPathInfos`.
Ref: https://github.com/apollographql/federation/blob/02ebb9ed09f157c4bcc00dbe27de6b57422b98cc/composition-js/src/validate.ts#L788

Reply:
There's more of an explanation in https://github.com/apollographql/federation/pull/3258 , but the gist is that the number of subgraph paths has a multiplicative effect that we want to watch out for, and this limit in particular is to avoid there being too many subgraph paths stored in the stack (which could cause the callee to run out of memory).